Setting the mrenclave to the rpc server context in the omni-executor

silva-fj opened the pull request:
As topic.

The `dcap_types` have been extracted from `common/primitives/core/src/teebag/sgx_verify/mod.rs` so we can import them in the omni-executor (there are some conflicts with the `webpki` crate).

silva-fj committed:
ts-tests initial setup

silva-fj committed:
setting up eslint

silva-fj committed:
changing dependencies and dev tool setup

silva-fj committed:
cleaning up example test

silva-fj committed:
updating exports from the parachain-api library

silva-fj committed:
setting up test utils (WIP)

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1306-add-integration-tests-for-the-omni-executor

silva-fj committed:
fixing context

silva-fj committed:
fix cli params

silva-fj committed:
adding missing NativeCall variant definition

silva-fj committed:
adding attestation crate

silva-fj committed:
removing attestation from the parentchain listener

silva-fj committed:
performing attestation when running executor-worker

silva-fj committed:
fixing responses from rpc methods

silva-fj committed:
fixing type definitions for extrinsic reports

silva-fj committed:
adding utilities to send plain requests

silva-fj committed:
adjusting type creators

silva-fj committed:
adding helpers

silva-fj committed:
setting up create_account_store integration test

silva-fj committed:
fixing generateVerificationMessage

silva-fj committed:
adding buildWeb3ValidationData util

silva-fj committed:
adding more type creators

silva-fj committed:
fixing initial nonce

silva-fj committed:
adding test for adding web3 account

silva-fj committed:
fixing native intent calls

silva-fj committed:
adding helper to fund accounts

silva-fj committed:
removing console.log

silva-fj committed:
adding test for publicize_account call

silva-fj committed:
removing unused imports

silva-fj committed:
adding test to remove accounts

silva-fj committed:
adding test for request intent

silva-fj committed:
cleaning up debugging logs

silva-fj committed:
refactor(attestation): remove unused dependencies

silva-fj committed:
adding test for set_permissions

silva-fj committed:
fixing clippy issues

silva-fj committed:
feat(teebag): move SGX DCAP types to dedicated module

Move all DCAP quote and report related types to a new module to improve code
organization and readability. This includes the DcapQuote, SgxReportBody,
SgxQuote and related structs. No functional changes, just code reorganization.

Types were moved from sgx_verify/mod.rs to the new dcap_types.rs module and
re-exported through teebag/mod.rs to maintain compatibility.

silva-fj committed:
updating omni-executor.manifest.template to include local directory

silva-fj committed:
export more types from heima primitives

silva-fj committed:
feat(rpc): add Debug derive for ExtrinsicReport struct

silva-fj committed:
feat(attestation): return MrEnclave from attestation flow

silva-fj committed:
fix(tee): set actual MrEnclave in RPC server context

Replace hardcoded MrEnclave value with the actual measurement obtained
from the attestation flow. This ensures proper enclave verification and
authentication in the RPC server.

silva-fj committed:
importing vec from alloc to keep compatibility

silva-fj committed:
fixing clippy issues

silva-fj committed:
adding TODO

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1322-set-the-actual-mrenclave-to-the-rpc-server-context

silva-fj committed:
fixing fmt issue

silva-fj committed:
fixing fmt issue

silva-fj committed:
cleaning up merging conflict issue

silva-fj committed:
refactor(attestation): update log message for MRENCLAVE

Update the log message format for MRENCLAVE value to be more consistent with
our logging standards. Removed a resolved TODO comment as the implementation
has been verified.

silva-fj committed:
fixing imports in sgx_verify tests

silva-fj committed:
Merge branch 'dev' into p-1322-set-the-actual-mrenclave-to-the-rpc-server-context

kziemianek commented on common/primitives/core/src/teebag/dcap_types.rs:
```diff
@@ -0,0 +1,301 @@
+use crate::{MrEnclave, MrSigner, QuotingEnclave, SgxBuildMode};
```
This file is missing a license header.

kziemianek commented on tee-worker/omni-executor/omni-executor.manifest.template:
```diff
@@ -42,6 +43,7 @@ sgx.trusted_files = [
 
 sgx.allowed_files = [
   "file:storage_db/",
+  "file:local/",
```
What's being stored here ?

silva-fj replied:
`log` and `keystore`. I have updated the manifest to protect the `storage_db` and the `keystore`

silva-fj committed:
feat(core): add copyright notice to dcap_types.rs

silva-fj committed:
feat: encrypt keystore and storage_db paths

Modify the manifest template to encrypt the storage_db and
local/keystore paths in the filesystem mounts configuration.
This ensures that sensitive data is protected when accessed
by the application running inside the secure enclave.

Removed unnecessary allowed_files section as this is now
handled by the encrypted mounts.
